modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/ExtensionStateTracker.java

```diff
@@ -74,29 +70,45 @@
      */
     List<ConfigurationInstanceProvider<?>> getConfigurationInstanceProviders()
     {
-        return ImmutableList.copyOf(configurationInstanceProviders.values());
+        return ImmutableList.copyOf(
+                configurationInstanceProviders.values()
+                        .stream()
+                        .map(wrapper -> wrapper.getConfigurationInstanceProvider())
+                        .collect(Collectors.toList()));
     }
 
     /**
      * Registers the creation of a new configuration instance
      *
-     * @param instanceKey           the key under which the instance was registered in the {@link MuleRegistry}
+     * @param providerName          the name of a registered {@link ConfigurationInstanceProvider}
+     * @param registrationName      the name on which the {@code configurationInstance} has been registered on the {@link MuleRegistry}
      * @param configurationInstance the configuration instance. Cannot be {@code null}
      * @param <C>                   the generic type for the {@code configurationInstance}
      */
-    <C> void registerConfigurationInstance(String instanceKey, C configurationInstance)
+    <C> void registerConfigurationInstance(String providerName, String registrationName, C configurationInstance)
```

Comment:
nop. Registration name is more clear. Instance name is more open to interpretation and confusion between that name and the name of the provider, etc